metrics/honeycomb.go

```diff
@@ -228,26 +226,23 @@ func (h *HoneycombMetrics) reportToHoneycommb(ctx context.Context) {
 				"api_host": ev.APIHost,
 				"dataset":  ev.Dataset,
 			}
-			h.countersLock.Lock()
```

Comment:
removing individual locks would make the entire function block for the whole reporting loop -- would that slow things down?

Reply:
I don't think it will have much of an affect. Most of the time we expect to be using read locks only (collect / updating existing metric values).

We use a write lock when adding new metrics (register).